### ISSUE TITLE
Support conversion of multiple enex files with preserving links

### ISSUE THREAD

**Issue by akosbalasko** [akosbalasko/yarle#292]
@pmottishaw had an idea (thank you for that) of supporting multiple enex files with link preservation. I think it would be a really great improvement (link preservation was not done by the old version of yarle, I simply forgot the scenario of linking notes inter-enex, shame on me).

**Comment by pmottishaw** [comment]
Thank you for adding the issue. What I would like is that the desktop app allows you to select multiple enex files not just a single enex. I think the command line version already has this feature, but have been unable to test it because I could not get the command line version to work. 

The linking is not a big problem. I have found that if I process all the enex from one Evernote notebook and then copy all the notes into the same folder and all the attachment folders into the same _resource folder all the links work with a few exceptions. The major exception is that if the note has an attachment with the same name as the note Obsidian will often link to the attachment when it should link to the note. There is an easy fix which is to choose to add the .md extension to links when configuring Yarle. The other exceptions seem to be having characters like > in the title (not sure).

In summary: the linking problem is solved by automating the merging of folders from the different enex files from the same notebook.

**Comment by stanislavkozlovski** [comment]
How does link preservation work with direct evernote links (e.g `[Life Trappings](https://www.evernote.com/shard/s515/nl/192780579/b56733a8-b750-068a-e559-abf65b52c0a2?title=Life%20Trappings)]`) ?

**Comment by akosbalasko** [comment]
Hi @stanislavkozlovski ,
Well, unfortunately these NotesGuids cannot be used as they are, because they are not exported with the note itself (the guids, not the links). So, it's a bit tricky, currenty I use the title to recognize the target note. Which is a bit dummy way, by at least something. 

A bit smarter way I'm thinking on would be to force a creation of TOC in Evernote as a prerequisite to be able to keep the links, and make sure that the labels of every note are unique. And if keep links is checked in Yarle, I would look for the TOC of the notebook and use that as a single point of trust, because it will contain that guids together with the note's title (by default it's the title, but they must be unique to prevent mixing the links up possibly).

**Comment by stanislavkozlovski** [comment]
Makes sense
